lib/utils/socketChainHelper.js

```diff
@@ -88,6 +90,27 @@ const processServerResponse = toString => function processServerResponseHandler(
 		}));
 	}
 
+	// takeScreenshot success
+	if (isTakeScreenshot && isSuccess) {
+		if (data.dataFormat === 'raw') {
+			return res.buffer;
+		} else if (data.dataFormat === 'base64') {
+			return res.buffer.toString('base64');
+		} else if (data.fileName) {
+			try {
+				fs.writeFileSync(data.fileName, res.buffer);
```

Comment:
Unless it's too much work, we should use async version here, so that JS thread is not blocked by filesystem operation